app/src/pages/homePage/home.jsx

```diff
@@ -6,21 +6,54 @@ import PostCmp from '../../components/post/post'
 import API from "../../utils/api";
 
 import { useEffect, useState } from "react";
-import { Navigate } from "react-router-dom";
+import { Navigate, useLocation } from "react-router-dom";
+
+import Cookies from "universal-cookie";
 
 export default function Home() {
 
-    const [navNewPost, setNaveNewPost] = useState(false)
+    const cookies = new Cookies(null, {path : '/'})
+    const user = cookies.get("currentUser")
 
+    const location = useLocation()
+
+    const [navNewPost, setNaveNewPost] = useState(false)
+    const [showAllPosts, setShowAllPosts] = useState(true)
     const [allPosts, setAllPosts] = useState([])
+    const [recommendations, setRecommendations] = useState([])
+    const [likedPosts, setLikedPosts] = useState([])
+
+
 
     useEffect(()=> {
         async function getPosts(){
             setAllPosts(await API.getAllPosts())
         }
 
+        async function getRecs(){
+            if (user.id){
+                setRecommendations(await API.getRecommendations(user.id))
+            }
+        }
+
+        async function getLiked(){
+            if (user.id){
+                const liked = await API.getLiked(user.id)
+                setLikedPosts(liked)
+            }
+        }
```

Comment:
you can define the functions `getPosts`, `getRecs`, and `getLiked` outside the `useEffect`!

you can also probably combine the `getRecs()` and `getLiked()` functions (or combine both them and the `getPosts` function), since you're using them all at the same time and not reusing!